crates/router/src/connector/fiserv/transformers.rs

```diff
@@ -147,6 +193,46 @@ impl TryFrom<&types::ConnectorAuthType> for FiservAuthType {
     }
 }
 
+#[derive(Default, Debug, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct FiservCancelRequest {
+    transaction_details: TransactionDetails,
+    merchant_details: MerchantDetails,
+    reference_transaction_details: ReferenceTransactionDetails,
+}
+
+impl TryFrom<&types::PaymentsCancelRouterData> for FiservCancelRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsCancelRouterData) -> Result<Self, Self::Error> {
+        let auth: FiservAuthType = FiservAuthType::try_from(&item.connector_auth_type)?;
+        let metadata = item
+            .connector_meta_data
+            .clone()
+            .ok_or(errors::ConnectorError::RequestEncodingFailed)?;
+        let session: SessionObject = metadata
+            .parse_value("SessionObject")
+            .change_context(errors::ConnectorError::RequestEncodingFailed)?;
```

Comment:
move this to utils and use it as `metadata.get_session_object()?`

Reply:
This is a connector specific custom `session Object` created in transformers itself, it cannot go under utils